Tests/AppTests/AnalyzerTests.swift

```diff
@@ -1192,6 +1203,81 @@ struct Command: Equatable, CustomStringConvertible, Hashable, Comparable {
 }
 
 
+// TODO: rename to Command or something else
+struct Command2: CustomStringConvertible {
+    var command: Cmd
+    var path: String
+
+    // TODO: find a name for this
+    enum Cmd {
+        case checkout
+        case clean
+        case clone(String)
+        case commitCount
+        case dumpPackage
+        case fetch
+        case firstCommitDate
+        case lastCommitDate
+        case getTags
+        case reset
+        case resetToBranch(String)
+        case showDate
+        case revisionInfo
+    }
+
+    init?(cmd: ShellOutCommand, path: String) {
+        let separator = "-"
+        self.path = path
+        switch cmd {
+            case _ where cmd.string.starts(with: "git checkout"):
+                self.command = .checkout
+            case .gitClean:
+                self.command = .clean
+            case _ where cmd.string.starts(with: "git clone"):
+                let url = String(cmd.string.split(separator: " ")
+                                    .filter { $0.contains("https://") }
+                                    .first!)
+                self.command = .clone(url)
+            case .gitCommitCount:
+                self.command = .commitCount
+            case .gitFetch:
+                self.command = .fetch
+            case .gitFirstCommitDate:
+                self.command = .firstCommitDate
+            case .gitLastCommitDate:
+                self.command = .lastCommitDate
+            case .gitTag:
+                self.command = .getTags
+            case .gitReset(hard: true):
+                self.command = .reset
+            case _ where cmd.string.starts(with: #"git reset "origin"#):
+                let branch = String(cmd.string.split(separator: " ")[2])
+                    .trimmingCharacters(in: .init(charactersIn: "\""))
+                self.command = .resetToBranch(branch)
+            case _ where cmd.string.starts(with: #"git show -s --format=%ct"#):
+                self.command = .showDate
+            case _ where cmd.string.starts(with: #"git log -n1 --format=format:"%H\#(separator)%ct""#):
+                self.command = .revisionInfo
+            case .swiftDumpPackage:
+                self.command = .dumpPackage
+            default:
+                return nil
+        }
+    }
```

Comment:
I'm not in love with the fact that this adds a bit of logic to the test mechanics itself. Like for instance parsing out the cloned url and fetched branch. OTOH it's making the dump actually readable.

We could still use the commands to make mocking easier but forego `Command2` (as it is named right now) and keep the dump as it is.